nautobot_golden_config/nornir_plays/config_intended.py

```diff
@@ -32,7 +35,22 @@
 InventoryPluginRegister.register("nautobot-inventory", NautobotORMInventory)
 LOGGER = logging.getLogger(__name__)
 
-jinja_env = engines["jinja"].env
+# Use a custom Jinja2 environment instead of Django's to avoid HTML escaping
+# Trim_blocks option defaulted to True to match nornir's default environment
+OPTION_LSTRIP_BLOCKS = False
+OPTION_TRIM_BLOCKS = True
+if PLUGIN_CFG.get("jinja_env_trim_blocks"):
+    OPTION_TRIM_BLOCKS = PLUGIN_CFG.get("jinja_env_trim_blocks")
+if PLUGIN_CFG.get("jinja_env_lstrip_blocks"):
+    OPTION_LSTRIP_BLOCKS = PLUGIN_CFG.get("jinja_env_lstrip_blocks")
+
+jinja_env = SandboxedEnvironment(
+    undefined=StrictUndefined,
+    trim_blocks=OPTION_TRIM_BLOCKS,
+    lstrip_blocks=OPTION_LSTRIP_BLOCKS,
+)
+# Retrieve filters from the Djanog jinja template engine
+jinja_env.filters = engines["jinja"].env.filters
```

Comment:
Is it possible we are missing anything that was in `engines["jinja"].env` before? I get that you are being more specific, and I presume in the past we only actually wanted the filters, but want to ensure behavior is not changing

Reply:
Hey Ken,
Prior to the nornir-nautobot change, it was not possible to pass the environment, only the filters were pushed through.

Since the environment could not be passed from nautobot-plugin-golden-config -> nornir-nautobot -> nornir-jinja2, this means that nornir-jinja2 was creating an environment inside the task.

https://github.com/nornir-automation/nornir_jinja2/blob/master/nornir_jinja2/plugins/tasks/template_file.py

``` 
    jinja_filters = jinja_filters or {}

    if jinja_env:
        env = jinja_env
        env.loader = FileSystemLoader(path)
    else:
        env = Environment(
            loader=FileSystemLoader(path), undefined=StrictUndefined, trim_blocks=True,
        )
    env.filters.update(jinja_filters)
```

The options that were set in this environment were:

`undefined: StrictUndefined`
`trim_blocks: True`

This means that it was also not possible to turn off the `trim_blocks` option if the user desired. We also could not turn on the `lstrip_blocks` option. The enabled options were controlled only by the defaults defined in the nornir-jinja2 library.

At one point, I did attempt to pass the `engine['jinja']` environment from Django, but the enabled options in that environment are not suited for rendering device configurations, and resulted in templates that had any special characters HTML escaped.

I also see that I have a typo in the comment on line 52 with a misspelling of 'Django'. I will push a commit to correct that.